### ISSUE TITLE
Why is rover body tilted forward?

### ISSUE THREAD

**Issue by JHPHELAN** [nasa-jpl/open-source-rover#58]
Why is rover body tilted forward?  I don't see that the actual Mars rovers had a forward tilt.  Is it just a happenstance of the choice/necessity of placement of the Differential Pivot and Rocker-Bogies?  I find no mention of it in the build documents.  I can reduce the tilt from ~14o to ~10o if I lengthen the turnbuckles from their minimum length to as long as I dare without them coming apart.  There is no mention in the build as to whether or how to adjust them.  

By the way, there is no mention of leaving any space between the main axle on the rocker-bogies and the ends of the differential pivot to avoid friction.  I left a mm or two.  I noticed there no lubricant washers next to the collar clamps along the main axle at the rocker-bogie arms & the body.

The last critical mechanical component, the 0.25" pillow bearing blocks are STILL on back-order in an apparent world-wide shortage!  Though promised next week, we're going to try carbon-fiber 3D printing the blocks and inserting our own bearings.  Will keep you posted how that goes.

Jim Phelan
Mars rover project lead
Houston Robotics

**Comment by ericjunkins** [comment]
Sorry for the lapse in responses here. The tilt in the body was an artifact of the standard parts to be found on servocity and their mounting options. As you have noticed you can control the angle of the body by varying the length of the turnbuckle lengths. In addition if you mount the attachments of the differential pivot back further on the rocker arm you can achieve the same effect. 

I have designed a 3D printed part to fix this, and will work on including it as an optional add-on for the built documentation, but I can post the information on it in this thread for now if you wish to use it.

The fix is a 3D printed piece the holds the 0.5 inch clamping hub in an appropriate spot such that when the turnbuckles are most of the way tightented the body becomes level. (STL file below)

[Diff pivot block.zip](https://github.com/nasa-jpl/open-source-rover/files/2568593/Diff.pivot.block.zip)

This piece utilizes heat set inserts to allow us to give threads to our 3D printed piece. Lulzbot has written a nice article about using them if you are unfimiliar. 

https://www.lulzbot.com/learn/tutorials/heat-set-inserts-tips-and-tricks

McMaster link to heat set inserts:
https://www.mcmaster.com/93365a130
(I recommend buying a couple different screw sizes while ordering these, they are invaluable to have on hand when prototyping and designing parts for 3D printing

Step1: Insert the heat-set brass inserts
![diff_pivot_block_1](https://user-images.githubusercontent.com/25470708/48304230-4318e880-e4cb-11e8-8ff6-fd49dfc8f27e.PNG)


Step2: Attach the clamping hub
![diff_pivot_block_2](https://user-images.githubusercontent.com/25470708/48304234-49a76000-e4cb-11e8-9f51-b83dc99c0444.PNG)

Step3: Mount the block on the rocker arm at shown location above the pillow bearing blocks
![diff_pivot_block_3](https://user-images.githubusercontent.com/25470708/48304236-4dd37d80-e4cb-11e8-8c0d-bea3275079a9.PNG)

**Comment by ericjunkins** [comment]
@JHPHELAN In addition I want to comment on your last part about printing bearing blocks. I do this frequently when I want to put bearings in systems. The biggest thing to understand is the tolerances of the 3D printer such that you are giving the bearing a nice tight press fit. 

It is important to realize where the load-path is of the robot and if this bearing block will be seeing load to determine if 3D printing is sufficient to house the bearing system. On the corner systems that bearing block must counteract the non-axial rotation of the corner wheel system. Under normal driving conditions this should be pretty small, the case of driving over aggressive obstacles however is where you start getting strain on this part. I believe that 3D printing should be sufficient for this part, and it certainly is a relatively easy part to replace later if it turns out it is not. I would assume ABS would be plenty strong, though a carbon printed part, printed in the correct orientation should be very robust.

I've attached a few examples of some bearing blocks I've printed with the servo city bearings. *Note I am printing these primarily on a Zortrax M200 printer, and might have different tolerances than your printer*

[bearing block examples.zip](https://github.com/nasa-jpl/open-source-rover/files/2568603/bearing.block.examples.zip)
